crates/common/tedge_config/src/tedge_config_cli/tedge_config.rs

```diff
@@ -663,6 +663,20 @@ define_tedge_config! {
             #[tedge_config(default(value = false))]
             #[doku(skip)] // Hide the configuration in `tedge config list --doc`
             built_in: bool,
+
+            reconnect_policy: {
+                /// The minimum time the built-in bridge will wait before reconnecting
+                #[tedge_config(example = "30", default(function = "|| Seconds(30)"))]
+                initial_interval: Seconds,
```

Comment:
I've used `Seconds` here, so we take an integer and use that as the number of seconds as that is consistent with other configurations. However, personally I think it would make more sense to use [humantime](https://docs.rs/humantime/latest/humantime/fn.parse_duration.html) to parse a duration value from the configuration.

This would allow a user both more granular precision if needed (`500ms` may make sense for this configuration), and clearer configuration of longer values (e.g. `5m` for 5 minutes instead of `300`).

Reply:
@reubenmiller @didier-wenzek what do you think is the best way to integrate this change with the existing values. Do we:
- Just change these configurations to support a human-time value
- Support humantime everywhere we currently use `Seconds`, but in a backwards-compatible manner (either by having a warning when converting an integer to a value of seconds, or by automatically rewriting `tedge.toml` to use the new format, like we did for changing some of the key names)

Reply:
> Just change these configurations to support a human-time value

Would could use "human-time" just for the settings related to this PR, and then look at using humantime by default for all intervals/durations in the config.

> Support humantime everywhere we currently use Seconds, but in a backwards-compatible manner (either by having a warning when converting an integer to a value of seconds, or by automatically rewriting tedge.toml to use the new format, like we did for changing some of the key names)

If no units are provided, I would just assume seconds (I'm not sure if adding a warning would be worthwhile). I would avoid rewriting the `tedge.toml`.

Reply:
Ideally, I would accept both:

- `initial_interval = 2` assuming this is a number of seconds
- `initial_interval = "5m"` where the human-time value is given as a string.

As this is backward compatible, why not apply this to all settings in `Seconds`.

Reply:
I've now modified the type to accept seconds or humantime formats. It preserves the input, so it won't rewrite seconds to humantime values, nor will it change the format of humantime values (so if someone prefers `1 hour` instead of `1h`, they can set that and it won't get clobbered when updating tedge config). I'm wondering:

1. Do we want to place limits on what humantime values we accept? For instance, there are units of "week", "month" and "year", all of which are likely longer than would ever be sensible for our existing configurations (and someone could plausibly write `1M` thinking it's a minute and actually it's a month). If these are sensible for some future configuration, I would argue accepting seconds in place of that is daft.
2. Do we want to convert existing seconds values to humantime values when people update other `tedge config set` commands? Aside from setting a value to `0`, it doesn't make a lot of sense in my mind for anyone to be using values that are implicitly seconds if they have a choice, and the format is (to me at least) intuitive. Equally, it might be confusing to have values change unexpectedly.

I've set the example and default values to use a short humantime format, e.g. `30s`/`5m`.

Reply:
> Do we want to place limits on what humantime values we accept? For instance, there are units of "week", "month" and "year", all of which are likely longer than would ever be sensible for our existing configurations (and someone could plausibly write 1M thinking it's a minute and actually it's a month). If these are sensible for some future configuration, I would argue accepting seconds in place of that is daft.

The limits are very config setting specific, so I would leave it up to the individual settings to define their own limits.

> Do we want to convert existing seconds values to humantime values when people update other tedge config set commands? Aside from setting a value to 0, it doesn't make a lot of sense in my mind for anyone to be using values that are implicitly seconds if they have a choice, and the format is (to me at least) intuitive. Equally, it might be confusing to have values change unexpectedly.

Can't we decouple the value in the tedge.toml from the stored value? If the value does not change, then why should the serialisation process change the value?

Reply:
> Do we want to place limits on what humantime values we accept?

I would not ask for `month` or `year` units if missing, but I see no value in adding code to remove these units. 

> Do we want to convert existing seconds values to humantime values when people update other tedge config set commands? 

I would not. As you said, it might be confusing to have values change unexpectedly.

Reply:
> Can't we decouple the value in the tedge.toml from the stored value? If the value does not change, then why should the serialisation process change the value?

Not easily, because the current process is:
1. Deserialize the entire `tedge.toml` into a `TedgeConfig` struct
2. Update the relevant configuration
3. Serialize the `TedgeConfig` struct to update the `tedge.toml`

We could plausibly modify things in the future to use something like [toml_edit](https://docs.rs/toml_edit/latest/toml_edit/) to do minimal, comment-preserving updates to the toml file, only affecting the key that is being changed, but right now this isn't possible.

Even without that though, the input to `tedge config set` isn't a TOML value. So in this case the approach that makes most sense is to parse the input to the config value's type and then (as a separate step) serialize it to a TOML value. As a result we would still need to store the input if we want to store the user provided values verbatim.